### PR TITLE
Runio scripts

### PR DESCRIPTION
**Pull request by DavertMik**
## Motivation/Description of the PR

Finally, I replaced my build scripts in PHP (see RoboFile.php).
To do this I had to build a new package [runio.js](https://github.com/consolidation/runio).

Available commands:

```
➜  ./runio.js
Usage:  <command> [options]

Options:
  -V, --version          output the version number
  -h, --help             output usage information

Commands:
  docs                    generate all docs (runs all docs:* commands in parallel)
  docs:plugins            generate documentation for plugins
  docs:external-helpers   generate documentation for helpers outside of main repo
  docs:helpers            generate documentation for helpers
  wiki                    publish wiki pages to website
  publish:site [user]     updates codecept.io website
  server                  run test server. Warning! PHP required!
  release                 publish a new release on npm. Update version in package.json!

```

Those can't be implemented via npm scripts, because for building docs we need:

* read all helpers
* loop over helpers replacing {{> placeholders }} with partials from `docs/webapi`
* generate documentation from helpers
* add header into *.md files

And so for other tasks as well.

Now all those commands can be executed via `./runio.js`:

```
# build docs
./runio docs
```

## Type of change

- [ ] Breaking changes
- [ ] New functionality
- [ ] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `robo docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: runio.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  },
+
+  async publishSite(user = 'davertmik') {
+    // updates codecept.io website
+    await processChangelog();
+    copy('docker/README.md', 'docs/docker.md');
+    await this.wiki();
+    await chdir('website', async () => {
+      await exec('npm install');
+      await npmRun('publish-gh-pages', cfg => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 50
           

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

File: runio.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    });
+  },
+
+
+  async server() {
+    // run test server
+    await Promise.all([
+      exec('php -S 127.0.0.1:8000 -t test/data/app'),
+      npmRun('json-server'),
+      ]
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected indentation of 4 spaces but found 6. (indent)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    });
+
+    writeToFile('docs/examples.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: examples')
+      cfg.line('title: Examples')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own examples to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Examples)')
+      cfg.textFromFile('website/wiki/Examples.md')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    });
+
+    writeToFile('docs/books.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: books')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('title: Books & Posts')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own books or posts to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Books-&-Posts)')
+      cfg.textFromFile('website/wiki/Books-&-Posts.md')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('id: community-helpers')
+      cfg.line('title: Community Helpers')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Share your helpers at our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Community-Helpers)')
+      cfg.line('')
+      cfg.textFromFile('website/wiki/Community-Helpers.md')
+    });
+
+    writeToFile('docs/examples.md', cfg => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+  },
+
+
+  async server() {
+    // run test server
+    await Promise.all([
+      exec('php -S 127.0.0.1:8000 -t test/data/app'),
+      npmRun('json-server'),
+      ]
+    )
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Unexpected newline before '('. (function-paren-newline)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('')
+      cfg.line('> Share your helpers at our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Community-Helpers)')
+      cfg.line('')
+      cfg.textFromFile('website/wiki/Community-Helpers.md')
+    });
+
+    writeToFile('docs/examples.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: examples')
+      cfg.line('title: Examples')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+      this.docsExternalHelpers(),
+    ]);
+
+  },
+
+  async docsPlugins() {
+    // generate documentation for plugins
+    await npx('documentation build lib/plugin/*.js -o docs/plugins.md -f md --shallow --markdown-toc=false --sort-order=alpha');
+    await replaceInFile('docs/plugins.md', cfg => {
+      cfg.replace(/^/,'---\nid: plugins\ntitle: Plugins\n---\n\n')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    )
+  },
+
+  async release() {
+    // publish a new release on npm. Update version in package.json!
+    const packageInfo = JSON.parse(fs.readFileSync('package.json'));
+    const version = packageInfo.version;
+    await this.docs();
+    await this.publishSite();
+    await this.git(cmd => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+
+    await writeToFile('docs/community-helpers.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: community-helpers')
+      cfg.line('title: Community Helpers')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Share your helpers at our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Community-Helpers)')
+      cfg.line('')
+      cfg.textFromFile('website/wiki/Community-Helpers.md')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+
+  async docsHelpers() {
+    // generate documentation for helpers
+    const files = fs.readdirSync('lib/helper').filter(f => path.extname(f) === '.js');
+
+    const partials = fs.readdirSync('docs/webapi').filter(f => path.extname(f) === '.mustache');
+    const placeholders = partials.map(file => `{{> ${path.basename(file, '.mustache')} }}`);
+    const templates = partials
+    .map(file => fs.readFileSync('docs/webapi/' + file).toString())
+    .map(template => template.replace(/^/m, '\n   * '));
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected indentation of 6 spaces but found 4. (indent)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own books or posts to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Books-&-Posts)')
+      cfg.textFromFile('website/wiki/Books-&-Posts.md')
+    });
+
+    writeToFile('docs/videos.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: videos')
+      cfg.line('title: Videos')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+
+module.exports = {
+  async docs() {
+    await Promise.all([
+      this.docsHelpers(),
+      this.docsPlugins(),
+      this.docsExternalHelpers(),
+    ]);
+
+  },
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Block must not be padded by blank lines. (padded-blocks)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+
+  async release() {
+    // publish a new release on npm. Update version in package.json!
+    const packageInfo = JSON.parse(fs.readFileSync('package.json'));
+    const version = packageInfo.version;
+    await this.docs();
+    await this.publishSite();
+    await this.git(cmd => {
+      cmd.tag(version);
+      cmd.push('origin master --tags')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.textFromFile('website/wiki/Books-&-Posts.md')
+    });
+
+    writeToFile('docs/videos.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: videos')
+      cfg.line('title: Videos')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own videos to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Videos)')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+  },
+
+  async docsHelpers() {
+    // generate documentation for helpers
+    const files = fs.readdirSync('lib/helper').filter(f => path.extname(f) === '.js');
+
+    const partials = fs.readdirSync('docs/webapi').filter(f => path.extname(f) === '.mustache');
+    const placeholders = partials.map(file => `{{> ${path.basename(file, '.mustache')} }}`);
+    const templates = partials
+    .map(file => fs.readFileSync('docs/webapi/' + file).toString())
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Unexpected string concatenation. (prefer-template)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+    await npx('documentation build lib/plugin/*.js -o docs/plugins.md -f md --shallow --markdown-toc=false --sort-order=alpha');
+    await replaceInFile('docs/plugins.md', cfg => {
+      cfg.replace(/^/,'---\nid: plugins\ntitle: Plugins\n---\n\n')
+    })
+  },
+
+  async docsExternalHelpers() {
+    // generate documentation for helpers outside of main repo
+    console.log('Building @codecepjs/detox helper docs');
+    let helper = 'Detox';
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: ['helper' is never reassigned. Use 'const' instead. (prefer-const)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 19]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('---')
+      cfg.line('id: books')
+      cfg.line('title: Books & Posts')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own books or posts to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Books-&-Posts)')
+      cfg.textFromFile('website/wiki/Books-&-Posts.md')
+    });
+
+    writeToFile('docs/videos.md', cfg => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 20]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('---')
+      cfg.line('id: videos')
+      cfg.line('title: Videos')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own videos to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Videos)')
+      cfg.textFromFile('website/wiki/Videos.md')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 21]
```diff
@@ -0,0 +1,195 @@
...
+    const sharedPlaceholders = sharedPartials.map(file =>`{{ ${path.basename(file, '.mustache')} }}`);
+    const sharedTemplates = sharedPartials
+      .map(file => fs.readFileSync('docs/shared/' + file).toString())
+      .map(template => '\n\n\n' + template)
+
+    for (const file of files) {
+      const name = path.basename(file,'.js');
+      console.log('Writing documentation for '+name);
+      copy(`lib/helper/${file}`,`docs/build/${file}`);
+      replaceInFile(`docs/build/${file}`, cfg => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 22]
```diff
@@ -0,0 +1,195 @@
...
+      cfg.line('---')
+      cfg.line('id: examples')
+      cfg.line('title: Examples')
+      cfg.line('---')
+      cfg.line('')
+      cfg.line('> Add your own examples to our [Wiki Page](https://github.com/Codeception/CodeceptJS/wiki/Examples)')
+      cfg.textFromFile('website/wiki/Examples.md')
+    });
+
+    writeToFile('docs/books.md', cfg => {
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected parentheses around arrow function argument having a body with curly braces. (arrow-parens)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 23]
```diff
@@ -0,0 +1,195 @@
...
+
+    const sharedPartials = fs.readdirSync('docs/shared').filter(f => path.extname(f) === '.mustache');
+    const sharedPlaceholders = sharedPartials.map(file =>`{{ ${path.basename(file, '.mustache')} }}`);
+    const sharedTemplates = sharedPartials
+      .map(file => fs.readFileSync('docs/shared/' + file).toString())
+      .map(template => '\n\n\n' + template)
+
+    for (const file of files) {
+      const name = path.basename(file,'.js');
+      console.log('Writing documentation for '+name);
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Operator '+' must be spaced. (space-infix-ops)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 24]
```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env node
+const fs = require('fs');
+const path = require('path');
+const { stopOnFail, chdir, git, copy, exec, replaceInFile, npmRun, npx, writeToFile, runio } = require('runio');
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Expected a line break before this closing brace. (object-curly-newline)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

File: runio.js [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+    if (!fs.existsSync('website/wiki/Home.md')) {
+      await git(fn => {
+        fn.cloneShallow('git@github.com:Codeception/CodeceptJS.wiki.git', 'website/wiki');
+      });
+    }
+    await chdir('website/wiki', () => git(cfg => cfg.pull('origin master')));
+
+    await writeToFile('docs/community-helpers.md', cfg => {
+      cfg.line('---')
+      cfg.line('id: community-helpers')
```
Comment (DavertMik) [inline runio.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4496293)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

File: docs/helpers/SeleniumWebdriver.md [hunk 26]
```diff
@@ -7,7 +7,7 @@ title: SeleniumWebdriver
 
 ## SeleniumWebdriver
```
Comment (PeterNgTr) [inline docs/helpers/SeleniumWebdriver.md]: Should be remove this helper, cause we don't use it anymore.

File: runio.js [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+
+  async server() {
+    // run test server. Warning! PHP required!
+    await Promise.all([
+      exec('php -S 127.0.0.1:8000 -t test/data/app'),
+      npmRun('json-server'),
+    ]);
+  },
+
+  async release() {
```
Comment (PeterNgTr) [inline runio.js]: don't you think it's good to have automatic versioning? We just pass, `minor` , `major` or `patch` and it is automatically update the package version.

File: docs/helpers/SeleniumWebdriver.md (hunk 26 above)
Comment (DavertMik) [reply docs/helpers/SeleniumWebdriver.md]: Ok!

File: runio.js (hunk 27 above)
Comment (DavertMik) [reply runio.js]: Good idea!

File: package.json [hunk 28]
```diff
@@ -33,8 +33,8 @@
     "build": "tsc -p ./",
     "json-server": "./node_modules/json-server/bin/index.js test/data/rest/db.json -p 8010 --watch -m test/data/rest/headers.js",
     "json-server:graphql": "node test/data/graphql/index.js",
-    "lint": "eslint bin/ examples/ lib/ test/ translations/ gulpfile.js",
-    "lint-fix": "eslint bin/ examples/ lib/ test/ translations/ gulpfile.js --fix",
+    "lint": "eslint bin/ examples/ lib/ test/ translations/ runio.js",
+    "lint-fix": "eslint bin/ examples/ lib/ test/ translations/ runio.js --fix",
     "docs": "robo docs",
```
Comment (elukoyanov) [inline package.json]: `robo` should be changed

File: .github/CONTRIBUTING.md [hunk 29]
```diff
@@ -114,15 +114,16 @@ Please try to add corresponding testcase to runner or unit.
...
 **Documentation for helpers is a part of a source code**.
-Whenever you need to update docs for a helper do it inside a .js file.
 
-In order to generate new documentation from source code run the following command with [Robo](https://robo.li/):
+> **Whenever you need to update docs for a helper do it inside a .js file.**
+
+After you updated docblock in JS file, generate markdown files with next command:
 
 ```
-robo docs:helpers
```
Comment (elukoyanov) [inline .github/CONTRIBUTING.md]: Maybe replace to `npm run docs` or something similar.

File: runio.js [hunk 30]
```diff
@@ -0,0 +1,196 @@
...
+  // user
+  changelog = changelog.replace(/\s@([\w-]+)/mg, ' **[$1](https://github.com/$1)**');
+
+  // issue
+  changelog = changelog.replace(/#(\d+)/mg, '[#$1](https://github.com/Codeception/CodeceptJS/issues/$1)');
+
+  // helper
+  changelog = changelog.replace(/\s\[(\w+)\]\s/mg, ' **[$1]** ');
+
+  writeToFile('docs/changelog.md', (cfg) => {
```
Comment (elukoyanov) [inline runio.js]: what is `cfg` for?

File: .github/CONTRIBUTING.md (hunk 29 above)
Comment (DavertMik) [reply .github/CONTRIBUTING.md]: there is no need in npm layer to do that.

File: package.json (hunk 28 above)
Comment (DavertMik) [reply package.json]: yes!

File: runio.js (hunk 30 above)
Comment (DavertMik) [reply runio.js]: this is the syntax in runio, with a callback function you configure the behavior of a task.
once I get more time I will write documentation for that

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

**Comment by PeterNgTr** [comment]
@DavertMik conflicting files 😞

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 31]
```diff
@@ -27,6 +27,6 @@ Applicable helpers:
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
 - [ ] Tests have been added
-- [ ] Documentation has been added (Run `robo docs`)
+- [ ] Documentation has been added (Run `./runio docs`)
```
Comment (elukoyanov) [inline .github/PULL_REQUEST_TEMPLATE.md]: `npm run docs`?

File: docs/helpers/Nightmare.md [hunk 32]
```diff
@@ -384,18 +348,18 @@ Additional arguments can be passed as well, while `done` function is always last
 
 ```js
 let val = await I.executeAsyncScript(function(url, done) {
-  // in browser context
-  $.ajax(url, { success: (data) => done(data); }
+// in browser context
```
Comment (elukoyanov) [inline docs/helpers/Nightmare.md]: All code examples in mustache fragments lost tabulation

File: docs/helpers/WebDriverIO.md [hunk 33]
```diff
@@ -916,10 +856,8 @@ I.pressKey(['Control','a']);
 
 #### Parameters
 
--   `key` ([string][10] \| [Array][17]&lt;[string][10]>) key or array of keys to press.
-    
-
-    {{> \_keys }}To make combinations with modifier and mouse clicks (like Ctrl+Click) press a modifier, click, then release it.
+-   `key` **([string][10] | [Array][17]&lt;[string][10]>)** key or array of keys to press.
+    {{> _keys }}To make combinations with modifier and mouse clicks (like Ctrl+Click) press a modifier, click, then release it.
```
Comment (elukoyanov) [inline docs/helpers/WebDriverIO.md]: should be `{{> _keys }}`  replaced by keys or link to keys?

File: docs/helpers/Detox.md [hunk 34]
```diff
@@ -82,7 +83,7 @@ I.appendField('name', 'davert');
 
 #### Parameters
 
--   `field` **CodeceptJS.LocatorOrString** 
+-   `field` **([string][6] \| [object][7])**
```
Comment (elukoyanov) [inline docs/helpers/Detox.md]: why `CodeceptJS.LocatorOrString ` was replaced to `([string][6] \| [object][7])` in this helper?

**Comment by elukoyanov** [comment]
I noticed, TestCafe helper is not listed in main menu -> Helpers.
Should it be added?

**Comment by elukoyanov** [comment]
I noticed, TestCafe helper is not listed in main menu -> Helpers.
Should it be added?

File: docs/helpers/Detox.md (hunk 34 above)
Comment (DavertMik) [reply docs/helpers/Detox.md]: No idea :)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

File: docs/helpers/Nightmare.md (hunk 32 above)
Comment (DavertMik) [reply docs/helpers/Nightmare.md]: Yes, that's true. I don't how to fix that in this moment

File: docs/helpers/WebDriverIO.md (hunk 33 above)
Comment (DavertMik) [reply docs/helpers/WebDriverIO.md]: Ok!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 31 above)
Comment (DavertMik) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done!

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- runio.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4496293&bid=15160429)

**Pull request merged**